### ISSUE TITLE
[PROPOSAL] Sunflower Land Radio

### ISSUE THREAD

**Issue by Rofrtd** [sunflower-land/sunflower-land#218]
This issue aims to explore a solution for adding music to Sunflower Land.

The main idea is to create a radio where we can add songs from community musicians.

### Main idea
Ideally the radio would have stations based on music styles that players could subscribe to. 
Radio subscription will be optional.
Each station would give a boost on a specific mechanic of the farm (to be discussed), for example:
   * Country music could give and extra 10% chance on getting max (5) wood when chopping trees
   * Rock music could keep goblins away for a couple of minutes each day.
   * Jazz...
   * etc..

We could also include giveaways/promotions/announcements using the radio, and only people really listening to the radio would be able to participate. 
Example: 
    * You listen on the radio that there is a hidden treasure on the snow map that you can try finding clicking on the map..
    * You listen on the radio that there is a massive infestation of mice coming, and you can prevent them to invading your farm by taking some action.
    * etc...

I think having a JUKEBOX somewhere on the map for the music options would be great. 

### MVP
We could implement a simple radio with all sorts of stiles for now, and give boosts to players who subscribe to the radio. 
The boost could change every week, for example:
   * Week 1 would make sunflower seeds 5% cheaper
   * Week 2 would keep the Goblins always for 30 minutes on a specific date and time.
   * etc...

### Discussion
Pricing for subscribing to the Sunflower Radio. 
    * It can't be too expensive, to be accessible to everyone. 
    * But it can't be too cheap, so we can give nice boosts to listeners.
    * A percentage of the subscriptions would go to the project.
   
How would we distribute the rewards back to the song writers?
    * Maybe we could divide equally the subscriptions (after the project fee) based on number of total songs owned per artist.
        * Example, if there are 10 songs on the radio and you wrote one of them you get 10% of it.
        
        
### Request
I would love to see some discussions/recommendations from the community here for the best ways to implement these ideas.

**Comment by Samuvieira07** [comment]
I really liked the proposal, you have 100% of my support! It would be nice to have a top 10 list, with prizes for the top 3

**Comment by adamhannigan** [comment]
Love the idea. Really cool that the music could give bonuses 🚀 

Playing a certain song to give a certain bonus is a cool way to gamify it. I.e. more upbeat music makes the plants grow faster 🥇 

Do you think people would subscribe for the music or for the functionality it gives? IMO, the work to build and support the feature and the functionality the music provides heavily outweighs the time it could take to write the music (2 days-1 week).

If a song is purchased/subscribed solely for the purpose of background music, I think 70-90% to the musician makes sense.

If a song is purchased for functionality, I think 10-30% for the musician is more reasonable.

For an MVP, changing the boosts each week might be a bit too much work. I would consider just having one boost per song and it remains the same forever.

**Comment by Rofrtd** [comment]
> 
I think if there were no bonuses people would still subscribe to the radio for the music only. But there would be a lot of people that wouldn't. So my approach was to target both sides, people who enjoy the music, and people who focus more on the game rewards.
So yes, I believe around 30% to the musician is reasonable having all those bonuses involved.

Yeah, I thought about having bonuses for each song too, but imagine when we have 30/50 songs on the radio and each one of them one different bonus. 
I don't think it would be viable at that point. That's why I mentioned having bonuses per style once we implement the full Radio.

**Comment by TerritorialBreak5** [comment]
Its a rly good idea the JukeBox, but it will be better, if you could craft it, (Jukebox as an NFT), and for craft we could use wood, stone, iron and other resources.

**Comment by j923sneaks** [comment]
> We could also include giveaways/promotions/announcements using the radio, and only people really listening to the radio would be able to participate.

Now this is cool!

> I think having a JUKEBOX somewhere on the map for the music options would be great.

Interesting approach. On one hand it lessens the clutter on the audio player. On the other hand it may not be intuitive to have 2 "action elements" for music. If there are multiple stations and the mechanics for choosing is complex, I definitely favor jukebox. Otherwise a simple left/right arrow w/ station name (`<- Jazz ->`) will suffice imo.

**Comment by adamhannigan** [comment]
@TerritorialBreak5 I like this idea. It encourages people to play the game and collect the resources they need for the feature 👌

**Comment by TerritorialBreak5** [comment]
> @TerritorialBreak5 I like this idea. It encourages people to play the game and collect the resources they need for the feature 👌

Yh, and after create the Jukebox, you can listen any station without paying more or after create the Jukebox, the farmer can pay for listen a specific station, like 1 Iron or other resource.

**Comment by Rofrtd** [comment]
> > @TerritorialBreak5 I like this idea. It encourages people to play the game and collect the resources they need for the feature 👌
> 
> Yh, and after create the Jukebox, you can listen any station without paying more or after create the Jukebox, the farmer can pay for listen a specific station, like 1 Iron or other resource.

Maybe with the Jukebox you can listen to the "free" tracks and announcements. But only get the boost/bonuses if you subscribe to a station.

**Comment by TerritorialBreak5** [comment]
> > > @TerritorialBreak5 I like this idea. It encourages people to play the game and collect the resources they need for the feature 👌
> > 
> > 
> > Yh, and after create the Jukebox, you can listen any station without paying more or after create the Jukebox, the farmer can pay for listen a specific station, like 1 Iron or other resource.
> 
> Maybe with the Jukebox you can listen to the "free" tracks and announcements. But only get the boost/bonuses if you subscribe to a station.

Agree!

**Comment by Z1kou** [comment]
I would suggest having a music station that is the norm( The music from V1 which is iconic for the game) , The addition for an optional radio which could be aquired lets say by aquest? , A traveling musician which is in dire need of fixing his music tools would hover around your town and if u provide him with ressources to fix his tools he would be paying u back with unlimited music or something like this , i Love this approach way better story wise , Ps: EMBRACE THE SHERRIFF, THE GOBLIN CATCHER.

**Comment by Pranav-yadav** [comment]
> The main idea is to create a radio where we can add songs from community musicians.

Wonderful ! building a strong community of more domains will definitely help in strengthening the SFL's ecosystem.👍

> If a song is purchased/subscribed solely for the purpose of background music, I think 70-90% to the musician makes sense.
> 
> If a song is purchased for functionality, I think 10-30% for the musician is more reasonable.

Definitely Adam, It'll not only help to build strong community along with gamified experience to players but also help to build a strong ecosystem.

> Its a rly good idea the JukeBox, but it will be better, if you could craft it, (Jukebox as an NFT), and for craft we could use wood, stone, iron and other resources.

Couldn't agree more :)

**Comment by Pranav-yadav** [comment]
> The main idea is to create a radio where we can add songs from community musicians.

Wonderful ! building a strong community of more domains will definitely help in strengthening the SFL's ecosystem.👍

> If a song is purchased/subscribed solely for the purpose of background music, I think 70-90% to the musician makes sense.
> 
> If a song is purchased for functionality, I think 10-30% for the musician is more reasonable.

Definitely Adam, It'll not only help to build strong community along with gamified experience to players but also help to build a strong ecosystem.

> Its a rly good idea the JukeBox, but it will be better, if you could craft it, (Jukebox as an NFT), and for craft we could use wood, stone, iron and other resources.

Couldn't agree more :)

**Comment by Pranav-yadav** [comment]
> The main idea is to create a radio where we can add songs from community musicians.

Wonderful ! building a strong community of more domains will definitely help in strengthening the SFL's ecosystem.👍

> If a song is purchased/subscribed solely for the purpose of background music, I think 70-90% to the musician makes sense.
> 
> If a song is purchased for functionality, I think 10-30% for the musician is more reasonable.

Definitely Adam, It'll not only help to build strong community along with gamified experience to players but also help to build a strong ecosystem.

> Its a rly good idea the JukeBox, but it will be better, if you could craft it, (Jukebox as an NFT), and for craft we could use wood, stone, iron and other resources.

Couldn't agree more :)

**Comment by Pranav-yadav** [comment]
> The main idea is to create a radio where we can add songs from community musicians.

Wonderful ! building a strong community of more domains will definitely help in strengthening the SFL's ecosystem.👍

> If a song is purchased/subscribed solely for the purpose of background music, I think 70-90% to the musician makes sense.
> 
> If a song is purchased for functionality, I think 10-30% for the musician is more reasonable.

Definitely Adam, It'll not only help to build strong community along with gamified experience to players but also help to build a strong ecosystem.

> Its a rly good idea the JukeBox, but it will be better, if you could craft it, (Jukebox as an NFT), and for craft we could use wood, stone, iron and other resources.

Couldn't agree more :)

**Comment by Rofrtd** [comment]
> I would suggest having a music station that is the norm( The music from V1 which is iconic for the game) , The addition for an optional radio which could be aquired lets say by aquest? , A traveling musician which is in dire need of fixing his music tools would hover around your town and if u provide him with ressources to fix his tools he would be paying u back with unlimited music or something like this , i Love this approach way better story wise , Ps: EMBRACE THE SHERRIFF, THE GOBLIN CATCHER.

I love the idea of having a quest to get the Jukebox. 
But having unlimited music after that I don't think is the best approach, as we need to reward musicians for their efforts on composing the tracks. 

But I do love the idea of having a gamified story behind the music/jukebox.

**Comment by stevewoody82** [comment]
Would like to see a partnership with Audius. Maybe even a sponsorship agreement could happen with them funding the project for their logo on the radio station. All artists could link their music into the Web3 Platform.

**Comment by kimberlite07** [comment]
Also maybe for those unsubscribed farmers, the radio can have ads like that of spotify so that it can earn as well :)

**Comment by Z1kou** [comment]
> > I would suggest having a music station that is the norm( The music from V1 which is iconic for the game) , The addition for an optional radio which could be aquired lets say by aquest? , A traveling musician which is in dire need of fixing his music tools would hover around your town and if u provide him with ressources to fix his tools he would be paying u back with unlimited music or something like this , i Love this approach way better story wise , Ps: EMBRACE THE SHERRIFF, THE GOBLIN CATCHER.
> 
> I love the idea of having a quest to get the Jukebox. But having unlimited music after that I don't think is the best approach, as we need to reward musicians for their efforts on composing the tracks.
> 
> But I do love the idea of having a gamified story behind the music/jukebox.

Yes I totally agree so let's say we do just that (how about we make a first quest in the tutorial of the game to introduce the musician he first just says hi ,and talks about his love of music I'm thinking u take the role of our ferryman musician and say a cool sweet msg in the first dialogue of the tutorial in the music section of it because let's admit it , it's gonna be a whole section now xd.)
TLDR
-Improve the tutorial in which everysection would be given a nice short concise explanation

-in the music part of the tutorial introduce the ferryman singer (we give him a character and maybe could play with events around him in the future again more story telling ;) )

THE GAME PLAN:
1- the ferryman singer is a cheerful person who would provide u with the classical V1 free version and just be there to brighten up your farm

2- the ferryman singer would be able to provide u with some spicy premium music (this is where we implement the creators rewarding system) in exchange of helping him fix his tools Everytime  . Every tool = new track with a certain reward = using ressources and money = rewarding the creator through what slf and money used for fixing the ferryman tool

-backstory
The ferryman is the embodiment of the music spirit, shares emotions throught his composing and connecting farmers with his might , be it love or hate sadness or excitement he can be it all ,using the power of thousands of musicians  he bless's you with songs and if your lucky he will bless your whole farm ,

Ps: Everytime you fix his tools (aka : buy a track of a creator, he references it by a quote in the msg and the creators name is displayed next to the music name with the effect it has Example: % growth for all crops planted in the day

**Comment by gabiqueiroga** [comment]
We could also use the jukebox/radio to warn us about some in game events. - they can be visual stuff happening in the map or warn about rabbits/rats stealing your crops. There are a lot of options and it is a way to burn some tokens

**Comment by gabiqueiroga** [comment]
still a WIP but..

https://user-images.githubusercontent.com/69090306/153087684-0b0e6c80-98db-4a2a-9f54-65b4029e7cec.mp4
